exercise_2/Graph.cpp

```diff
@@ -0,0 +1,77 @@
+#include "Graph.h"
+#include <iostream>
+
+Graph::Graph(int(&a)[N][N])
+{
+	for (int i = 0; i < N; ++i)
+	{
+		for (int j = 0; j < N; ++j)
+		{
+			arr[i][j] = a[i][j];
+		}
+	}
+}
+
+Graph::~Graph()
+{
+}
+
+void  Graph::printGraph()
+{
+	for (int i = 0; i < N; ++i)
+	{
+		for (int j = 0; j < N; ++j)
+		{
+			std::cout << arr[i][j] << " ";
+		}
+		std::cout << std::endl;
+	}
+}
+
+void Graph::printPath(int a[N], int start)
+{
+	for (int i = 0; i < N; i++)
+	{
+		std::cout << "From " << start << " to " << i << "  " << a[i] << " " << std::endl;
+	}
+}
+
+void  Graph::findShortestPath(int start = 0)
+{
+	int path[N]{ 0 };
```

Comment:
Rather use standard stl containers then raw arrays. Please check:
std::map, std::vector, std::list and so on.